merlin/study/celerymanager.py

```diff
@@ -0,0 +1,230 @@
+}
+
+
+class CeleryManager:
+    def __init__(self, query_frequency: int = 60, query_timeout: float = 0.5, worker_timeout: int = 180):
+        """
+        Initializer for Celery Manager
+        @param int query_frequency:     The frequency at which workers will be queried with ping commands
+        @param float query_timeout:     The timeout for the query pings that are sent to workers
+        @param int worker_timeout:      The sum total(query_frequency*tries) time before an attempt is made to restart worker.
+        """
+        self.redis_connection = self.get_worker_status_redis_connection()
+        self.query_frequency = query_frequency
+        self.query_timeout = query_timeout
+        self.worker_timeout = worker_timeout
+
+    @staticmethod
+    def get_worker_status_redis_connection():
+        """
+        Get the redis connection for info regarding the worker and manager status.
+        """
+        return CeleryManager.get_redis_connection(1)
+
+    @staticmethod
+    def get_worker_args_redis_connection():
+        """
+        Get the redis connection for info regarding the args used to generate each worker.
+        """
+        return CeleryManager.get_redis_connection(2)
+
+    @staticmethod
+    def get_redis_connection(db_num):
+        """
+        Generic redis connection function to get the results backend redis server with a given db number increment.
+        :param int db_num:      Increment number for the db from the one provided in the config file.
+
+        :return Redis:          Redis connections object that can be used to access values for the manager.
+        """
+        from merlin.config.configfile import CONFIG
+        from merlin.config.results_backend import get_backend_password
+
+        password_file = CONFIG.results_backend.password
+        try:
+            password = get_backend_password(password_file)
+        except IOError:
+            password = CONFIG.results_backend.password
+        return redis.Redis(
+            host=CONFIG.results_backend.server,
+            port=CONFIG.results_backend.port,
+            db=CONFIG.results_backend.db_num + db_num,  # Increment db_num to avoid conflicts
+            username=CONFIG.results_backend.username,
+            password=password,
+            decode_responses=True,
+        )
+
+    def get_celery_workers_status(self, workers):
+        """
+        Get the worker status of a current worker that is being managed
+        :param CeleryManager self:      CeleryManager attempting the stop.
+        :param list workers:            Workers that are checked.
+
+        :return dict:                   The result dictionary for each worker and the response.
+        """
+        from merlin.celery import app
+
+        celery_app = app.control
+        ping_result = celery_app.ping(workers, timeout=self.query_timeout)
+        worker_results = {worker: status for d in ping_result for worker, status in d.items()}
+        return worker_results
```

Comment:
Will `ping_results` only contain workers that are running? If ping doesn't get a response for a worker within the time limit what happens?